### ISSUE TITLE
Alt exits terminal mode

### ISSUE THREAD

**Issue by kuator** [neovim/neovim#16220]
### Neovim version (nvim -v)

NVIM v0.6.0-dev+545-g519d8deb0

### Vim (not Nvim) behaves the same?

I don't know, I don't use Vim

### Operating system/version

DISTRIB_DESCRIPTION="Linux Mint 20.2 Uma"

### Terminal name/version

Konsole Version 19.12.3

### $TERM environment variable

xterm-256color

### Installation

build from source

### How to reproduce the issue

```
:lua vim.cmd[[tnoremap <esc> <C-\><C-n>]]
:terminal
Type alt-b or any shortcut involving <alt>, and it will exit terminal mode
```

### Expected behavior

Mapping `<esc>` in `terminal-mode` doesn't affect `<alt>` mappings in any way

### Actual behavior

Typing any `<alt>` prefixed key mapping exits terminal mode

**Comment by zeertzjq** [comment]
See `:h map-alt-keys`

**Comment by zeertzjq** [comment]
Oh, this indeed seems to be a problem. IDK if there is a good solution to this.

**Comment by zeertzjq** [comment]
BTW, Vim behaves the same.

**Comment by zeertzjq** [comment]
See also #15869.

**Comment by kuator** [comment]
@zeertzjq I tried to reproduce this with neovim 0.5.1, and it doesn't have this problem. Alt mappings in terminal work as expected.

**Comment by zeertzjq** [comment]
Yes, because #15869 wasn't fixed in 0.5.1.

**Comment by zeertzjq** [comment]
I'm trying to fix this.

**Comment by zeertzjq** [comment]
I'm trying to fix this in #16222
